### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by ISSOtm**
[The documentation currently online](https://docs.rs/png/0.16.8/png/index.html) seems to largely assume familiarity with the PNG spec. Since I have minimal knowledge of the spec, this has given me trouble. I need a fairly low-level interface, since I am trying to port to Rust a program sensitive to whether a PNG is paletted or not.

This PR is intended to improve documentation, including to be more noob-friendly (as much as possible, anyway); I have made a few modifications based off my experience, and noted other possible changes below. If you have ideas, please tell me, and I will add them!

- [ ] A code comment mentions that some `ColorType`/`BitDepth` combinations are impossible. Maybe this should be pointed out in the docs?
- [ ] The C code block in `Transformations`'s doc does not have type for its constants, and it's not clear (to the reader) whether this code comes from this crate, or `libpng`. Further, what are they intended for?
- [ ] Maybe some more examples? [This is how I'm using the lib](https://github.com/ISSOtm/rsgbgfx/blob/52b9ccbc88f30a91f86085143212b701f0cca342/src/img/png.rs#L15), fwiw. How to use the various fields of `Info`, for example, I had to determine through trial and error.

### CODE REVIEW

File: src/common.rs [hunk 1]
```diff
@@ -474,12 +483,12 @@ impl Info {
...
     }
 
-    /// Returns the bits per pixel
+    /// Returns the number of bits per pixel.
     pub fn bits_per_pixel(&self) -> usize {
         self.color_type.samples() * self.bit_depth as usize
     }
 
-    /// Returns the bytes per pixel
+    /// Returns the number of bytes per pixel.
```
Comment (fintelia) [inline src/common.rs:491]: This should probably say that the number is rounded up to the nearest byte. Though to be honest I'm not quite sure why we have this function at all. When would you actually want to do rounding when computing pixel size?

**Comment by fintelia** [comment]
>  The C code block in Transformations's doc does not have type for its constants, and it's not clear (to the reader) whether this code comes from this crate, or libpng. Further, what are they intended for?

Those are the transformations from libpng which aren't yet supported by this crate, but which may be in the future. The syntax is actually for the bitflags macro (from the crate of the same name).

File: src/common.rs (hunk 1 above)
Comment (HeroicKatora) [reply src/common.rs:491]: The bytes per channel are rounded up. For 4bpp all channels are still in individual bytes.

**Review by HeroicKatora — APPROVED** [review]

**Comment by HeroicKatora** [comment]
In general I do not think the library should go too far into the png specification. The documentation should be written with the right terminology and in a style that ensures one can quickly find the relevant portions and paragraphs but replicating it in different wording is, in my opinion, not a good approach. For example, the allowed color combinations are due to the specification and while maybe being slightly more obvious to duplicate the table from the specification, it also gives the appearance that the library does something that diverges from the specification or does not support all combinations. That can be highly confusing as well.

> Maybe some more examples? This is how I'm using the lib, fwiw. How to use the various fields of Info, for example, I had to determine through trial and error.

Sure, that seems good. Most of the methods lack dedicated examples so short snippets would be appreciated.

**Comment by HeroicKatora** [comment]
Thanks! Feel free to open more PRs with additional improvements, as pointed out in your open questions.

**Comment by ISSOtm** [comment]
Oh, I thought I'd add more documentation to this PR before merging it. :P

To avoid repeating the spec, and encourage users to read it, a solution may be to pepper the documentation with links to relevant spec paragraphs? Could also be useful as citation-like reference, again to provide guidance to the user. (For example, I thought paletted images couldn't have alpha, until someone pointed out the `tRNS` chunk to me.)

**Comment by HeroicKatora** [comment]
Yes, link and section numbers would also be helpful.

**Pull request merged**